scripts/e2e-ci.js

```diff
@@ -0,0 +1,114 @@
+const yargs = require('yargs')
+const { spawn, exec } = require('child_process')
+const { promisify } = require('util')
+
+const pexec = promisify(exec)
+
+// Setup command line args
+// prettier-ignore
+const argv = yargs
+  .option('name', {
+    alias: 'n',
+    description: 'Name of the e2e project to test.',
+    demandOption: true,
+    requiresArg: true,
+    type: 'string'
+  })  
+  .option('type', {
+    alias: 't',
+    choices: ['next', 'react'],
+    default: 'next',
+    description: 'Type of app.',
+    type: 'string',
+  })
+  .option('port', {
+    alias: 'p',
+    default: '4200',
+    description: 'Port number of the app.',
+    type: 'number',
+  })
+  .option('folder', {
+    alias: 'f',
+    description: 'Dist folder of the app. Only needed for React apps.',
+    type: 'string',
+  })
+  .option('base-path', {
+    alias: 'b',
+    default: '/',
+    description: 'Base path of the app if it is deployed to a sub-folder.',
+    requiresArg: true,
+    type: 'string'
+  }) 
+  .option('ci', {
+    alias: 'c',
+    description: 'Flag to indicate if the script is being run from CI platform.',
+    type: 'boolean',
+  })
+  .help()
+  .alias('help', 'h')
+  .argv
+
+// Validate that the folder option is set when type=react
+if (argv.type === 'react' && !argv.folder) {
+  throw new Error('Missing required option: folder')
+}
+
+// Strip '-e2e' ending to get the target app name
+const target = argv.name.replace('-e2e', '')
+
+// We run the e2e tests on production built application
+// prettier-ignore
+const CMD = {
+  BUILD: `yarn nx run ${target}:build:production --skip-nx-cache`,
+  SERVE: [
+    'node',
+    [
+      'scripts/static-serve.js',
+      '-p', argv.port,
+      '-d', argv.folder,
+      '-b', argv['base-path']
+    ]
+  ],
+  TEST: `yarn nx run ${argv.name}:e2e:production --headless --production --base-url http://localhost:${argv.port}${
+    argv.ci ? 
+      ` --record --group=${argv.name}` 
+    : ''}`,
+}
+
+const main = async () => {
+  console.log(`Building ${target}...`)
+  await pexec(CMD.BUILD)
```

Comment:
await with then and catch? why?

Also probably means that an error in build won't stop the run but continue to serve/test

Reply:
Yes agreed this is stupid. I'm changing this.